src/fs.ts

```diff
@@ -0,0 +1,258 @@
+ * itself.
+ * @param jsonValue - The JSON-like value to write to the file. Make sure that
+ * JSON.stringify can handle it.
+ * @param options - The options to this function.
+ * @param options.prettify - Whether to format the JSON as it is turned into a
+ * string such that it is broken up into separate lines (using 2 spaces as
+ * indentation).
+ * @param options.stringifier - The stringifier to use. Defaults to `JSON`.
+ * @param options.stringifier.stringify - A function that stringifies JSON.
+ * @returns The object corresponding to the parsed JSON file, typed against the
+ * struct.
+ * @throws An error with a stack trace if writing fails in any way.
+ */
+export async function writeJsonFile(
+  filePath: string,
+  jsonValue: Json,
+  {
+    stringifier = JSON,
+    prettify = false,
+  }: {
+    stringifier?: {
+      stringify: typeof JSON.stringify;
+    };
+    prettify?: boolean;
+  } = {},
+): Promise<void> {
+  try {
+    await fs.promises.mkdir(path.dirname(filePath), { recursive: true });
+    const json = prettify
+      ? stringifier.stringify(jsonValue, null, '  ')
+      : stringifier.stringify(jsonValue);
+    await fs.promises.writeFile(filePath, json);
+  } catch (error) {
+    throw wrapError(error, `Could not write JSON file '${filePath}'`);
+  }
+}
+
+/**
+ * Tests the given path to determine whether it represents a file.
+ *
+ * @param filePath - The path to a (supposed) file on the filesystem.
+ * @returns A promise for true if the file exists or false otherwise.
+ * @throws An error with a stack trace if reading fails in any way.
+ */
+export async function fileExists(filePath: string): Promise<boolean> {
+  try {
+    const stats = await fs.promises.stat(filePath);
+    return stats.isFile();
+  } catch (error) {
+    if (isErrorWithCode(error) && error.code === 'ENOENT') {
+      return false;
+    }
+
+    throw wrapError(error, `Could not determine if file exists '${filePath}'`);
+  }
+}
+
+/**
+ * Tests the given path to determine whether it represents a file.
+ *
+ * @param directoryPath - The path to a (supposed) directory on the filesystem.
+ * @returns A promise for true if the file exists or false otherwise.
+ * @throws An error with a stack trace if reading fails in any way.
+ */
+export async function directoryExists(directoryPath: string): Promise<boolean> {
+  try {
+    const stats = await fs.promises.stat(directoryPath);
+    return stats.isDirectory();
+  } catch (error) {
+    if (isErrorWithCode(error) && error.code === 'ENOENT') {
+      return false;
+    }
+
+    throw wrapError(
+      error,
+      `Could not determine if file exists '${directoryPath}'`,
+    );
+  }
+}
+
+/**
+ * Creates the given directory along with any directories leading up to the
+ * directory. If the directory already exists, this is a no-op.
+ *
+ * @param directoryPath - The path to the desired directory.
+ * @returns What `fs.promises.mkdir` returns.
```

Comment:
What does it return? 😅

Reply:
Ha 😅 I've removed the return entirely here: [939e954](https://github.com/MetaMask/utils/pull/148/commits/939e95440b2dd0e5740a4be9ab029cc6c6988f9e)